flows-advanced/using_state_hooks/emit_event_on_completion.py

```diff
@@ -0,0 +1,42 @@
+"""This example uses `emit_event` to trigger a deployment when a flow run completes.
```

Comment:
A short README describing the two approaches and their tradeoffs would be good to have.